src/date.rs

```diff
@@ -16,12 +16,29 @@ use rkyv::{Archive, Deserialize, Serialize};
 use crate::format::Locale;
 #[cfg(any(feature = "alloc", feature = "std", test))]
 use crate::format::{DelayedFormat, Item, StrftimeItems};
-use crate::naive::{self, IsoWeek, NaiveDate, NaiveTime};
+use crate::naive::{self, IsoWeek, NaiveDate, NaiveTime, NaiveWeek};
 use crate::offset::{TimeZone, Utc};
 use crate::oldtime::Duration as OldDuration;
+use crate::traits::Weeklike;
 use crate::DateTime;
 use crate::{Datelike, Weekday};
 
+/// A week represented by a [`NaiveWeek`] + [Offset](crate::offset::Offset).
+#[derive(Debug)]
+pub struct Week<Tz: TimeZone> {
```

Comment:
So, after reviewing some other PRs I'm on the fence about this concept of types that don't point to a specific time but do reference a timezone. Since the type represents a time span, the timezone offset could conceivably be different at the start of the span vs the end of the span, which we cannot represent accurately.

How important is having this type included to you? Could we make do with just the `NaiveWeek`?

Reply:
@djc  - It makes sense to me, starting with `NaiveWeek` sounds fair. I just updated the PR simplifying the implementation a little bit.